Assignment4/course_materials/notebooks/task07.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1WqED9C82DGh6lb1Iuw2YtpDULNJlv8H-
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2023-2024/master/Assignment4/course_materials"
+
+"""First let's read the RDF file"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+# TO DO
+# Visualize the results
+q1 = """SELECT ?subClass
+WHERE {
+  ?subClass <http://www.w3.org/2000/01/rdf-schema#subClassOf> <http://somewhere#LivingThing> .
+}"""
+for r in g.query(q1):
+ print(r[0])
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+# TO DO
+# Visualize the results
+
+# Q1 filter out subClasses
+q1 = """SELECT ?x ?z
+WHERE {
+  ?x ?z <http://somewhere#Person> .
+  FILTER (?z != <http://www.w3.org/2000/01/rdf-schema#subClassOf>)
+}"""
+print('Query 1')
+for r in g.query(q1):
+ print(r[0])
+
+# OR Q2 list types of Person
+
+q2 = """SELECT ?x ?z
+WHERE {
+  ?x <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://somewhere#Person> .
+
+}"""
+print('Query 2')
+for r in g.query(q2):
+ print(r[0])
+
+"""**TASK 7.3: List all individuals of "Person" or "Animal" and all their properties including their class with RDFLib and SPARQL. You do not need to list the individuals of the subclasses of person**
+
+"""
+
+# TO DO
+# Visualize the results
+q1 = """SELECT ?x ?z ?class
+WHERE {
+  ?x ?property ?class .
+  FILTER (?property != <http://www.w3.org/2000/01/rdf-schema#subClassOf>) .
+  FILTER (?class = <http://somewhere#Person> || ?class = <http://somewhere#Animal>) .
+  ?x ?z ?y
+}"""
+
+for r in g.query(q1):
+ print(r[0], r[1], r[2])
+
+"""**TASK 7.4:  List the name of the persons who know Rocky**"""
+
+# TO DO
+# Visualize the results
+q1 = """SELECT ?fn
+WHERE {
+  ?x ?property ?class .
+  FILTER (?property != <http://www.w3.org/2000/01/rdf-schema#subClassOf>) .
+  FILTER (?class = <http://somewhere#Person>) .
+  ?x <http://xmlns.com/foaf/0.1/knows> ?y .
+  FILTER (?y = <http://somewhere#RockySmith>) .
```

Comment:
Remove filters for this query. If you do ?x ?property ?class, you are basically retrieving the full graph. No query will work in a realistic scenario